agent/internal/taskoutput/directory.go

```diff
@@ -106,5 +113,13 @@ type directoryHandler interface {
 	run(context.Context) error
 }
 
+// directoryHandlerOpts contains options to be passed into each directory handler implementation initialization.
+type directoryHandlerOpts struct {
+	output       *taskoutput.TaskOutput
+	taskOpts     taskoutput.TaskOptions
+	redactorOpts redactor.RedactionOptions
+	grpcConn     *grpc.ClientConn
```

Comment:
Nit: can we call this `oTelConn`? Just in case some future handler will require a different gRPC connection.